core-java-modules/core-java-date-operations-4/src/test/java/com/baeldung/compareyesterday/CompareYesterdayUnitTest.java

```diff
@@ -0,0 +1,119 @@
+package com.baeldung.compareyesterday;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.time.Instant;
+import java.time.LocalDate;
+import java.time.temporal.ChronoUnit;
+import java.util.Calendar;
+import java.util.Date;
+
+import org.joda.time.DateTime;
+import org.joda.time.DateTimeZone;
+import org.junit.jupiter.api.Test;
+
+public class CompareYesterdayUnitTest {
+
+    @Test
+    void givenYesterdayDate_whenCompareWithCalendar_returnTrue() {
+        // To simulate yesterday
+        Date date = new Date(System.currentTimeMillis() - (1000 * 60 * 60 * 24));
+        Calendar testCalendar = Calendar.getInstance();
+        testCalendar.setTime(date);
+
+        Calendar yesterdayDate = Calendar.getInstance();
```

Comment:
Note about the naming of the variables for the tests: I find it confusing for them to be named testDate and yesterdayDate, as these do not connotate which is the given value and which is the value under test. This confusion seems to have crept into the article draft as well. I think we would be better served with more conventional JUnit test variable names or prefixes, like expected and actual.